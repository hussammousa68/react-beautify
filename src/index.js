// @flow

// Components

export { default as DragDropContext } from './view/drag-drop-context';
export { default as Droppable } from './view/droppable';
export { default as Draggable } from './view/draggable';

// Utils

export { resetServerContext } from './view/drag-drop-context';

// Public flow types

export type {
  Id,
  TypeId,
  DraggableId,
  DroppableId,
  MovementMode,
  DragStart,
  DragUpdate,
  DropResult,
  HandleProvided,
  Announce,
  DraggableLocation,
  OnBeforeDragStartHandle,
  OnDragStartHandle,
  OnDragUpdateHandle,
  OnDragEndHandle,
} from './types';

// Droppable
export type {
  Provided as DroppableProvided,
  StateSnapshot as DroppableStateSnapshot,
  DroppableProps,
} from './view/droppable/droppable-types';

// Draggable
export type {
  Provided as DraggableProvided,
  StateSnapshot as DraggableStateSnapshot,
  DropAnimation,
  DraggableProps,
  DraggableStyle,
  DraggingStyle,
  NotDraggingStyle,
} from './view/draggable/draggable-types';

// DragHandle
export type { DragHandleProps } from './view/drag-handle/drag-handle-types';
