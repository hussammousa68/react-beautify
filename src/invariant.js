// @flow
const isProduction: boolean = process.env.NODE_ENV === 'production';
const prefix: string = 'Invariant failed';

export class RbdInvariant extends Error {}

// A copy-paste of tiny-invariant but with a custom error type
// Throw an error if the condition fails
export function invariant(condition: mixed, message?: string) {
  if (condition) {
    return;
  }

  if (isProduction) {
    // In production we strip the message but still throw
    throw new Error(prefix);
  } else {
    // When not in production we allow the message to pass through
    // *This block will be removed in production builds*
    throw new Error(`${prefix}: ${message || ''}`);
  }
}
