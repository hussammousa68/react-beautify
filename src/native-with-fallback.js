// @flow
type Map<T> = {
  [key: string]: T,
};

// @babel/runtime-corejs2 will replace Object.values
export function values<T>(map: Map<T>): T[] {
  // https://github.com/facebook/flow/issues/2221
  // $FlowFixMe - Object.values currently does not have good flow support
  return Object.values(map);
}

export function find<T>(list: Array<T>, predicate: T => boolean): ?T {
  if (list.find) {
    return list.find(predicate);
  }
  // Using a for loop so that we can exit early
  // alternative: list.filter(predicate)[0]
  for (let i = 0; i < list.length; i++) {
    if (predicate(list[i])) {
      return list[i];
    }
  }
  // Array.prototype.find returns undefined when nothing is found
  return undefined;
}

export function findIndex<T>(list: Array<T>, predicate: T => boolean): number {
  return list.findIndex
    ? list.findIndex(predicate)
    : list.indexOf(find(list, predicate));
}
