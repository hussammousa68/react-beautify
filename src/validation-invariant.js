// @flow
import { invariant } from './invariant';
import { warning } from './dev-warning';

let last: ?number = null;

export function validationInvariant(condition: mixed, message: string) {
  if (condition) {
    return;
  }

  const current: number = Date.now();
  if (last != null && current - last < 100) {
    warning(`
      Invariant failed.
      Not throwing error to prevent infinite loop.

      Message: > ${message}
    `);
    return;
  }
  last = current;
  invariant(condition, message);
}
