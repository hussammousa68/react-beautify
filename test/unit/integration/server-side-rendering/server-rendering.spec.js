// @flow
/**
 * @jest-environment node
 */
/* eslint-disable no-console */
import React from 'react';
import { renderToString, renderToStaticMarkup } from 'react-dom/server';
import invariant from 'tiny-invariant';
import { resetServerContext } from '../../../../src';
import App from './app';

const consoleFunctions: string[] = ['warn', 'error', 'log'];

beforeEach(() => {
  // Reset server context between tests to prevent state being shared between them
  resetServerContext();
  consoleFunctions.forEach((name: string) => {
    jest.spyOn(console, name);
  });
});

afterEach(() => {
  consoleFunctions.forEach((name: string) => {
    console[name].mockRestore();
  });
});

const expectConsoleNotCalled = () => {
  consoleFunctions.forEach((name: string) => {
    console[name].not.toHaveBeenCalled();
  });
};

// Checking that the browser globals are not available in this test file
invariant(
  typeof window === 'undefined' && typeof document === 'undefined',
  'browser globals found in node test',
);

it('should support rendering to a string', () => {
  const result: string = renderToString(<App />);

  expect(result).toEqual(expect.any(String));
  expect(result).toMatchSnapshot();
  expectConsoleNotCalled();
});

it('should support rendering to static markup', () => {
  const result: string = renderToStaticMarkup(<App />);

  expect(result).toEqual(expect.any(String));
  expect(result).toMatchSnapshot();
  expectConsoleNotCalled();
});

it('should render identical content when resetting context between renders', () => {
  const firstRender = renderToString(<App />);
  const nextRenderBeforeReset = renderToString(<App />);
  expect(firstRender).not.toEqual(nextRenderBeforeReset);

  resetServerContext();
  const nextRenderAfterReset = renderToString(<App />);
  expect(firstRender).toEqual(nextRenderAfterReset);
  expectConsoleNotCalled();
});
