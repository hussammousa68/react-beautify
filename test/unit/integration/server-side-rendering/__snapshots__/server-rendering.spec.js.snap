// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should support rendering to a string 1`] = `"<div data-rbd-droppable-id=\\"drop-1\\" data-rbd-droppable-context-id=\\"0\\"><div data-rbd-draggable-context-id=\\"0\\" data-rbd-draggable-id=\\"drag-1\\" tabindex=\\"0\\" data-rbd-drag-handle-draggable-id=\\"drag-1\\" data-rbd-drag-handle-context-id=\\"0\\" aria-labelledby=\\"rbd-lift-instruction-0\\" draggable=\\"false\\">Drag me!</div></div>"`;

exports[`should support rendering to static markup 1`] = `"<div data-rbd-droppable-id=\\"drop-1\\" data-rbd-droppable-context-id=\\"0\\"><div data-rbd-draggable-context-id=\\"0\\" data-rbd-draggable-id=\\"drag-1\\" tabindex=\\"0\\" data-rbd-drag-handle-draggable-id=\\"drag-1\\" data-rbd-drag-handle-context-id=\\"0\\" aria-labelledby=\\"rbd-lift-instruction-0\\" draggable=\\"false\\">Drag me!</div></div>"`;
